apps/api/test/functional/verify-email.spec.ts

```diff
@@ -0,0 +1,158 @@
+      expect(userSetting?.emailVerified).toBe(true);
+    });
+
+    it("sets emailVerified to false when Auth0 responds with email_verified=false", async () => {
+      const { user, token } = await setup({ emailVerified: false });
+
+      const response = await app.request("/v1/verify-email", {
+        method: "POST",
+        headers: {
+          authorization: `Bearer ${token}`,
+          "content-type": "application/json"
+        },
+        body: JSON.stringify({
+          data: {
+            email: user.email
+          }
+        })
+      });
+
+      expect(response.status).toBe(200);
+      const result = await response.json();
+      expect(result).toMatchObject({
+        data: {
+          emailVerified: false
+        }
+      });
+
+      const userSetting = await UserSetting.findOne({ where: { userId: user.userId } });
+      expect(userSetting?.emailVerified).toBe(false);
+    });
+
+    it("throws 404 when user is not found in auth0", async () => {
+      const { user, token } = await setup({ emailVerified: false });
+      const auth0Service = container.resolve(Auth0Service);
+      jest.spyOn(auth0Service, "getUserByEmail").mockResolvedValue(null);
+
+      const response = await app.request("/v1/verify-email", {
+        method: "POST",
+        headers: {
+          authorization: `Bearer ${token}`,
+          "content-type": "application/json"
+        },
+        body: JSON.stringify({
+          data: {
+            email: user.email
+          }
+        })
+      });
+
+      expect(response.status).toBe(404);
+    });
+
+    it("throws 401 if user is not authenticated", async () => {
+      const { user } = await setup({ emailVerified: false });
+
+      const response = await app.request("/v1/verify-email", {
+        method: "POST",
+        body: JSON.stringify({
+          data: {
+            email: user.email
+          }
+        })
+      });
+
+      expect(response.status).toBe(401);
+    });
+
+    it("throws 403 when email is not the same as the current user's email", async () => {
+      const { token } = await setup({ emailVerified: false });
+
+      const response = await app.request("/v1/verify-email", {
+        method: "POST",
+        headers: {
+          authorization: `Bearer ${token}`,
+          "content-type": "application/json"
+        },
+        body: JSON.stringify({
+          data: {
+            email: faker.internet.email()
+          }
+        })
+      });
+
+      expect(response.status).toBe(403);
+    });
+
+    async function setup(input: { emailVerified: boolean }) {
+      const walletTestingService = new WalletTestingService(app);
+      const { user, token } = await walletTestingService.createRegisteredUser({
+        email_verified: input.emailVerified
+      });
+
+      const auth0Service = container.resolve(Auth0Service);
+      jest.spyOn(auth0Service, "getUserByEmail").mockImplementation(async (email: string) => {
```

Comment:
I'm thinking this should be mocked via request interception as this is a functional? wdyt @baktun14 @stalniy ?